### PR TITLE
DOC: Update the CPU build options document

### PR DESCRIPTION
**Pull request by ghost**
ref: gh-25786

Updated example commands to use ``pip`` instead of ``setup.py``. Meson doesn't seem to have a command to specify CPU options for ``build_ext`` and ``build_clib`` individually.

The build log (in the ``Build Report`` section) of this document also needs to be updated (gh-24865), but this has not been done yet.

[skip cirrus] [skip azp] [skip actions]

### CODE REVIEW

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks for working on this @F3eQnxN3RriK. Overall it looks good, I have some detailed comments only.

**Comment by F3eQnxN3RriK** [comment]
@rgommers Thank you for your review comments. Fixed them.

File: doc/source/reference/simd/build-options.rst [hunk 1]
```diff
@@ -26,39 +26,16 @@ that target certain CPU features:
...
-    cd /path/to/numpy
-    python setup.py build_clib --cpu-baseline="avx2 fma3" install --user
-
-You can also customize CPU/build options through PIP command::
+To customize CPU/build options::
 
-    pip install --no-use-pep517 --global-option=build \
-    --global-option="--cpu-baseline=avx2 fma3" \
-    --global-option="--cpu-dispatch=max" ./
+    pip install -Csetup-args=-Dcpu-baseline="avx2 fma3" -Csetup-args=-Dcpu-dispatch="max" .
```
Comment (rgommers) [inline doc/source/reference/simd/build-options.rst:38]:
```suggestion
pip install . -Csetup-args=-Dcpu-baseline="avx2 fma3" -Csetup-args=-Dcpu-dispatch="max"
```

File: doc/source/reference/simd/build-options.rst [hunk 2]
```diff
@@ -177,64 +154,64 @@ Behaviors
...
   ``-march=native`` or ``-xHost`` or ``/QxHost`` is enabled through environment variable
-  `CFLAGS`::
+  ``CFLAGS``::
 
     export CFLAGS="-march=native"
-    python setup.py install --user
+    pip install .
     # is equivalent to
-    python setup.py build --cpu-baseline=native install --user
+    pip install -Csetup-args=-Dcpu-baseline=native .
```
Comment (rgommers) [inline doc/source/reference/simd/build-options.rst:190]:
```suggestion
    pip install . -Csetup-args=-Dcpu-baseline=native
```

File: doc/source/reference/simd/build-options.rst [hunk 3]
```diff
@@ -177,64 +155,64 @@ Behaviors
...
-          python setup.py build --cpu-baseline="sse sse2 sse3 ssse3 sse41 popcnt sse42"
+          pip wheel --config-settings=setup-args=-Dcpu-baseline="sse sse2 sse3 ssse3 sse41 popcnt sse42" .
 
-- ``--cpu-dispatch`` does not combain any of implied CPU features,
+- ``cpu-dispatch`` does not combain any of implied CPU features,
   so you must add them unless you want to disable one or all of them::
 
     # Only dispatches AVX2 and FMA3
-    python setup.py build --cpu-dispatch=avx2,fma3
+    pip wheel --config-settings=setup-args=-Dcpu-dispatch=avx2,fma3 .
```
[the remark this answers is not in the document]
Comment (rgommers) [reply doc/source/reference/simd/build-options.rst]: Ah, I see why I was confused, that was "install local dir", not the end of a sentence. Usually these come before flags for clarity.

**Review by rgommers — APPROVED** [review]
Looks good now, thanks @F3eQnxN3RriK

**Pull request merged**